From 5355997fc3f6ef137d65df3dfcb234c7df5a49d4 Mon Sep 17 00:00:00 2001
From: Matt Clarkson <mattyclarkson@gmail.com>
Date: Fri, 15 Mar 2024 16:37:13 +0000
Subject: [PATCH] Switch `purego` build tags to `cgo`

`rules_go` does not provide `purego` build tags when `pure = "on"`.

`cgo` build tag is the inverse so we can update the source files
to conditionally compile on `cgo`.
---
 dh/csidh/fp511_amd64.s              | 3 ++-
 dh/csidh/fp511_noasm.go             | 4 ++--
 dh/sidh/internal/p434/arith_amd64.s | 3 ++-
 dh/sidh/internal/p503/arith_amd64.s | 3 ++-
 dh/sidh/internal/p751/arith_amd64.s | 3 ++-
 dh/x25519/curve_amd64.go            | 4 ++--
 dh/x25519/curve_amd64.s             | 3 ++-
 dh/x25519/curve_noasm.go            | 4 ++--
 dh/x448/curve_amd64.go              | 4 ++--
 dh/x448/curve_amd64.s               | 3 ++-
 dh/x448/curve_noasm.go              | 4 ++--
 ecc/fourq/fp_amd64.go               | 4 ++--
 ecc/fourq/fp_amd64.s                | 3 ++-
 ecc/fourq/fp_noasm.go               | 4 ++--
 ecc/fourq/fq_amd64.go               | 4 ++--
 ecc/fourq/fq_amd64.s                | 3 ++-
 ecc/fourq/fq_noasm.go               | 4 ++--
 ecc/fourq/point_amd64.go            | 4 ++--
 ecc/fourq/point_amd64.s             | 3 ++-
 ecc/fourq/point_generic.go          | 4 ++--
 ecc/fourq/point_noasm.go            | 4 ++--
 ecc/p384/arith_amd64.s              | 3 ++-
 math/fp25519/fp_amd64.go            | 4 ++--
 math/fp25519/fp_amd64.s             | 3 ++-
 math/fp25519/fp_noasm.go            | 4 ++--
 math/fp448/fp_amd64.go              | 4 ++--
 math/fp448/fp_amd64.s               | 3 ++-
 math/fp448/fp_noasm.go              | 4 ++--
 28 files changed, 56 insertions(+), 44 deletions(-)

diff --git a/dh/csidh/fp511_amd64.s b/dh/csidh/fp511_amd64.s
index 612a0c5..2a2405c 100644
--- a/dh/csidh/fp511_amd64.s
+++ b/dh/csidh/fp511_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 #include "textflag.h"
 
diff --git a/dh/csidh/fp511_noasm.go b/dh/csidh/fp511_noasm.go
index 72b2efd..6018821 100644
--- a/dh/csidh/fp511_noasm.go
+++ b/dh/csidh/fp511_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package csidh
 
diff --git a/dh/sidh/internal/p434/arith_amd64.s b/dh/sidh/internal/p434/arith_amd64.s
index 785997b..25675d5 100644
--- a/dh/sidh/internal/p434/arith_amd64.s
+++ b/dh/sidh/internal/p434/arith_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!noasm
+//go:build amd64 && cgo
+// +build amd64,cgo,!noasm
 
 #include "textflag.h"
 
diff --git a/dh/sidh/internal/p503/arith_amd64.s b/dh/sidh/internal/p503/arith_amd64.s
index 2845cbd..7260c28 100644
--- a/dh/sidh/internal/p503/arith_amd64.s
+++ b/dh/sidh/internal/p503/arith_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!noasm
+//go:build amd64 && cgo
+// +build amd64,cgo,!noasm
 
 #include "textflag.h"
 
diff --git a/dh/sidh/internal/p751/arith_amd64.s b/dh/sidh/internal/p751/arith_amd64.s
index 59f5fa6..488cc97 100644
--- a/dh/sidh/internal/p751/arith_amd64.s
+++ b/dh/sidh/internal/p751/arith_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!noasm
+//go:build amd64 && cgo
+// +build amd64,cgo,!noasm
 
 #include "textflag.h"
 
diff --git a/dh/x25519/curve_amd64.go b/dh/x25519/curve_amd64.go
index 8a3d54c..07c8224 100644
--- a/dh/x25519/curve_amd64.go
+++ b/dh/x25519/curve_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package x25519
 
diff --git a/dh/x25519/curve_amd64.s b/dh/x25519/curve_amd64.s
index b772318..6309e42 100644
--- a/dh/x25519/curve_amd64.s
+++ b/dh/x25519/curve_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 #include "textflag.h"
 
diff --git a/dh/x25519/curve_noasm.go b/dh/x25519/curve_noasm.go
index 07fab97..551eb87 100644
--- a/dh/x25519/curve_noasm.go
+++ b/dh/x25519/curve_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package x25519
 
diff --git a/dh/x448/curve_amd64.go b/dh/x448/curve_amd64.go
index a062266..2e3805d 100644
--- a/dh/x448/curve_amd64.go
+++ b/dh/x448/curve_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package x448
 
diff --git a/dh/x448/curve_amd64.s b/dh/x448/curve_amd64.s
index 810aa9e..e18a840 100644
--- a/dh/x448/curve_amd64.s
+++ b/dh/x448/curve_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 #include "textflag.h"
 
diff --git a/dh/x448/curve_noasm.go b/dh/x448/curve_noasm.go
index 3755b7c..d5aef33 100644
--- a/dh/x448/curve_noasm.go
+++ b/dh/x448/curve_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package x448
 
diff --git a/ecc/fourq/fp_amd64.go b/ecc/fourq/fp_amd64.go
index da1ba4b..c2f63fc 100644
--- a/ecc/fourq/fp_amd64.go
+++ b/ecc/fourq/fp_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package fourq
 
diff --git a/ecc/fourq/fp_amd64.s b/ecc/fourq/fp_amd64.s
index 9543bd2..0336a12 100644
--- a/ecc/fourq/fp_amd64.s
+++ b/ecc/fourq/fp_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo,cgo
 
 #include "textflag.h"
 #include "fp_amd64.h"
diff --git a/ecc/fourq/fp_noasm.go b/ecc/fourq/fp_noasm.go
index aa35f52..586475c 100644
--- a/ecc/fourq/fp_noasm.go
+++ b/ecc/fourq/fp_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fourq
 
diff --git a/ecc/fourq/fq_amd64.go b/ecc/fourq/fq_amd64.go
index af880cd..579e85e 100644
--- a/ecc/fourq/fq_amd64.go
+++ b/ecc/fourq/fq_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package fourq
 
diff --git a/ecc/fourq/fq_amd64.s b/ecc/fourq/fq_amd64.s
index 415fc13..52adf79 100644
--- a/ecc/fourq/fq_amd64.s
+++ b/ecc/fourq/fq_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo,cgo
 
 #include "fq_amd64.h"
 
diff --git a/ecc/fourq/fq_noasm.go b/ecc/fourq/fq_noasm.go
index 1c7716a..c8af974 100644
--- a/ecc/fourq/fq_noasm.go
+++ b/ecc/fourq/fq_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fourq
 
diff --git a/ecc/fourq/point_amd64.go b/ecc/fourq/point_amd64.go
index aeb7272..376e190 100644
--- a/ecc/fourq/point_amd64.go
+++ b/ecc/fourq/point_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package fourq
 
diff --git a/ecc/fourq/point_amd64.s b/ecc/fourq/point_amd64.s
index f4dd5dd..90fbe77 100644
--- a/ecc/fourq/point_amd64.s
+++ b/ecc/fourq/point_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo,cgo
 
 #include "go_asm.h"
 #include "fq_amd64.h"
diff --git a/ecc/fourq/point_generic.go b/ecc/fourq/point_generic.go
index 792cd31..beef6c2 100644
--- a/ecc/fourq/point_generic.go
+++ b/ecc/fourq/point_generic.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fourq
 
diff --git a/ecc/fourq/point_noasm.go b/ecc/fourq/point_noasm.go
index 22c2695..ab476a4 100644
--- a/ecc/fourq/point_noasm.go
+++ b/ecc/fourq/point_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fourq
 
diff --git a/ecc/p384/arith_amd64.s b/ecc/p384/arith_amd64.s
index 5f53c63..33957d1 100644
--- a/ecc/p384/arith_amd64.s
+++ b/ecc/p384/arith_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64,!noasm
+//go:build amd64 && cgo
+// +build amd64,cgo,!noasm
 
 #include "textflag.h"
 
diff --git a/math/fp25519/fp_amd64.go b/math/fp25519/fp_amd64.go
index 057f0d2..fb71dbf 100644
--- a/math/fp25519/fp_amd64.go
+++ b/math/fp25519/fp_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package fp25519
 
diff --git a/math/fp25519/fp_amd64.s b/math/fp25519/fp_amd64.s
index 5c4aedd..d8e1eaa 100644
--- a/math/fp25519/fp_amd64.s
+++ b/math/fp25519/fp_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 #include "textflag.h"
 #include "fp_amd64.h"
diff --git a/math/fp25519/fp_noasm.go b/math/fp25519/fp_noasm.go
index 26ca4d0..1fb7c7a 100644
--- a/math/fp25519/fp_noasm.go
+++ b/math/fp25519/fp_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fp25519
 
diff --git a/math/fp448/fp_amd64.go b/math/fp448/fp_amd64.go
index 6a12209..f4097f9 100644
--- a/math/fp448/fp_amd64.go
+++ b/math/fp448/fp_amd64.go
@@ -1,5 +1,5 @@
-//go:build amd64 && !purego
-// +build amd64,!purego
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 package fp448
 
diff --git a/math/fp448/fp_amd64.s b/math/fp448/fp_amd64.s
index 435addf..643f07e 100644
--- a/math/fp448/fp_amd64.s
+++ b/math/fp448/fp_amd64.s
@@ -1,4 +1,5 @@
-// +build amd64
+//go:build amd64 && cgo
+// +build amd64,cgo
 
 #include "textflag.h"
 #include "fp_amd64.h"
diff --git a/math/fp448/fp_noasm.go b/math/fp448/fp_noasm.go
index a62225d..f2366c5 100644
--- a/math/fp448/fp_noasm.go
+++ b/math/fp448/fp_noasm.go
@@ -1,5 +1,5 @@
-//go:build !amd64 || purego
-// +build !amd64 purego
+//go:build !amd64 || !cgo
+// +build !amd64 !cgo
 
 package fp448
 
